python/lib/physiological.py

```diff
@@ -549,6 +577,162 @@ def insert_event_file(self, event_data, event_file, physiological_file_id,
+                                      physiological_annotation_file
+         :type annotation_data      : list
+        :param annotation_file      : name of the annotation file
+         :type annotation_file      : str
+        :param physiological_file_id: PhysiologicalFileID to link the event info to
+         :type physiological_file_id: int
+        :param blake2               : blake2b hash of the task event file
+         :type blake2               : str
+        """
+
+        annotationFileID = self.db.insert(
+            table_name   = 'physiological_annotation_file',
+            column_names = ('PhysiologicalFileID', 'FileType',  'FilePath'),
+            values       = (physiological_file_id, 'tsv',  annotation_file),
+            get_last_id  = True
+        )
+
+        annotation_fields = (
+            'AnnotationFileID', 'AnnotationParameterID', 'Onset', 'Duration',
+            'AnnotationLabelID', 'Channel', 'AbsoluteTime', 'Description'
+        )
+        annotation_values = []
+        for row in annotation_data:
+            optional_fields = (
+                'channels', 'absolute_time', 'description'
+            )
+            for field in optional_fields:
+                if field not in row.keys():
+                    row[field] = None
+
+            if not row['duration'].isdecimal():
+                row['duration'] = None
+
+            if not row['onset'].isdecimal():
+                row['onset'] = None
+
+            if "n/a" in row['channels']:
+                row['channels'] = None
+
+            labelID = self.db.pselect(
+                query="SELECT AnnotationLabelID FROM physiological_annotation_label WHERE LabelName = %s "
+                "LIMIT 1",
+                args=(row['label'],)
+            )
+            labelID = labelID[0]['AnnotationLabelID']
+
+            if not labelID:
+                labelID = self.db.insert(
+                    table_name   = 'physiological_annotation_label',
+                    column_names = ('LabelName',),
+                    values       = (row['label'],),
+                    get_last_id  = True
+                )
+
+            paramID = self.db.pselect(
+                query="SELECT AnnotationParameterID "
+                "FROM physiological_annotation_parameter p "
+                "JOIN physiological_annotation_file f ON f.AnnotationFileID = p.AnnotationFileID "
+                "WHERE f.PhysiologicalFileID = %s "
+                "LIMIT 1",
+                args=(physiological_file_id,)
+            )
+            paramID = paramID[0]['AnnotationParameterID']
+
+            values_tuple = (
+                str(annotationFileID),
+                paramID,
+                row['onset'],
+                row['duration'],
+                labelID,
+                row['channels'],
+                row['absolute_time'],
+                row['description']
+            )
+            annotation_values.append(values_tuple)
+
+        self.db.insert(
```

Comment:
move this in a file lib/database_lib/physiologicalannotationinstance.py where all queries about physiological_annotation_instance.py would go.